src/coreclr/jit/codegenxarch.cpp

```diff
@@ -1010,7 +1010,36 @@ void CodeGen::genCodeForBinary(GenTreeOp* treeNode)
     // the same code as above
     else if (op2reg == targetReg)
     {
-        noway_assert(GenTree::OperIsCommutative(oper));
+        unsigned lclNum1 = (unsigned)-1;
+        unsigned lclNum2 = (unsigned)-2;
+
+#ifdef DEBUG
+        if (op1->OperIsLocalRead())
+        {
+            lclNum1 = op1->AsLclVarCommon()->GetLclNum();
+        }
+        else if (op1->IsCopyOrReload())
+        {
+            GenTree* copySrc = op1->AsCopyOrReload()->gtGetOp1();
+            if (copySrc->OperIsLocalRead())
+            {
+                lclNum1 = copySrc->AsLclVarCommon()->GetLclNum();
+            }
+        }
```

Comment:
This could be simpler as:
```suggestion
        GenTree* op1skip = op1->gtSkipReloadOrCopy();
        if (op1skip->OperIsLocalRead())
        {
            lclNum1 = op1skip->AsLclVarCommon()->GetLclNum();
        }
```

same for op2.

Reply:
ah, I forgot about `gtSkipReloadOrCopy()`. Let me change it.